File: problems/chapter12/nicetak/abc121_c.jl
function main()
    N, M = parse.(Int, split(readline()))
    AB = [parse.(Int, split(readline())) for _ in 1:N]

    sort!(AB)
    ans = 0
    for i in 1:N
        A, B = AB[i][1], AB[i][2]
        if M ≤ B
            ans += A * M
            break
        else
            ans += A * B
            M -= B
        end
    end

    println(ans)
end

main()

File: problems/chapter12/Ysi/abc121_c.py
def main():
    N, M = map(int, input().split())
    ab = []
    for _ in range(N):
        a, b = map(int, input().split())
        ab.append([a, b])

    ab_sorted = sorted(ab, key=lambda x: x[0])
    
    num = M
    ans = 0
    for i in range(N):
        a, b = ab_sorted[i]
        if num > b:
            num -= b
            ans += a*b
        else:
            ans += a*num
            num = 0
    
    print(ans)

if __name__=='__main__':
    main()